apps/common/controllers/add-another-address-loop.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+
+const _ = require('lodash');
+const BaseController = require('./base');
+
+module.exports = class AddAnotherAddressController extends BaseController {
+  constructor(options) {
+    super(options);
+    if (options.locals.field) {
+      this.field = options.locals.field;
+    } else {
+      // eslint-disable-next-line no-console
+      console.error('No field set in locals');
+    }
+  }
+
+  locals(req, res) {
+    const locals = super.locals(req, res);
+    const addresses = req.sessionModel.get(`${this.field}Addresses`);
+    const hasAddresses = req.sessionModel.get(`${this.field}Addresses`) ? true : false;
+    const items = _.map(addresses, (value, key) => ({
+      id: key,
+      address: value.address
+    }));
+    return Object.assign({}, locals, {
+      items,
+      hasAddresses
+    });
+  }
+
+  get(req, res, callback) {
+    if (req.params.action === 'delete' && req.params.id) {
+      return this.removeItem(req, res, this.field);
```

Comment:
we don't need to pass this.field as the remove method can now access it from the instance